internal/provider/kubernetes/testdata/in/referencegrant-experimental-crd.yaml

```diff
@@ -3,7 +3,7 @@ kind: CustomResourceDefinition
 metadata:
   annotations:
     api-approved.kubernetes.io: https://github.com/kubernetes-sigs/gateway-api/pull/1086
-    gateway.networking.k8s.io/bundle-version: v0.5.0
+    gateway.networking.k8s.io/bundle-version: v0.6.0-dev
```

Comment:
I upgraded testdata of GWAPI manifests with main branch, to pass the kube e2e tests.